ui/src/main/js/page/stocks/index.js

```diff
@@ -0,0 +1,180 @@
+		).then(stockCollection => {
+				return client({
+					method: 'GET',
+					path: stockCollection.entity._links.profile.href,
+					headers: {'Accept': 'application/schema+json'}
+				}).then(schema => {
+					this.schema = schema.entity;
+					this.links = stockCollection.entity._links;
+					return stockCollection;
+				});
+		}).then(stockCollection => {
+			this.page = stockCollection.entity.page;
+			return stockCollection.entity._embedded.stocks.map(stock =>
+					client({
+						method: 'GET',
+						path: stock._links.self.href
+					})
+			);
+		}).then(stockPromises => {
+			return when.all(stockPromises);
+		}).then(stocks => {
+			this.setState({
+				page: this.page,
+				stocks: stocks,
+				attributes: Object.keys(this.schema.properties),
+				pageSize: pageSize,
+				links: this.links
+			});
+		});
+	}
+
+    onNavigate(navUri) {
+        this.searchForStockCollection(navUri);
+    }
+
+  updatePageSize(pageSize) {
+    if (pageSize !== this.state.pageSize) {
+      this.loadFromServer(pageSize);
+    }
+  }
+
+  onSearch(searchCriteria) {
+    if(searchCriteria.toUpperCase().indexOf('LON:')>-1) {
+        this.searchBySymbol(searchCriteria);
+    } else {
+        this.searchByCompanyName(searchCriteria);
+    }
+  }
+
+  searchByCompanyName(companyName) {
+      const navUri = root+'/stocks/search/findByCompanyNameStartsWith?companyName='+companyName.toUpperCase();
+      this.searchForStockCollection(navUri);
+  }
+
+  searchForStockCollection(navUri) {
+      client({
+          method: 'GET',
+          path: navUri
+      }).then(stockCollection => {
+          this.links = stockCollection.entity._links;
+          this.page = stockCollection.entity.page;
+          return stockCollection.entity._embedded.stocks.map(stock =>
+                  client({
+                      method: 'GET',
+                      path: stock._links.self.href
+                  })
+          );
+      }).then(stockPromises => {
+          return when.all(stockPromises);
+      }).then(stocks => {
+          this.setState({
+              page: this.page,
+              stocks: stocks,
+              attributes: Object.keys(this.schema.properties),
+              pageSize: this.state.pageSize,
+              links: this.links
+          });
+      });
+  }
+
+  searchBySymbol(symbol) {
+      const navUri = root+'/stocks/search/findBySymbol?symbol='+symbol.toUpperCase();
+      client({
+          method: 'GET',
+          path: navUri
+      }).then(stock => {
+          this.setState({
+              page: this.page,
+              stocks: [stock],
+              pageSize: this.state.pageSize,
+              links: this.links
+          });
+      });
+  }
+
+  refreshCurrentPage(message) {
+    let updatedStock = message.body;//JSON.parse(message.body);
+  	follow(client, root, [{
+  		rel: 'stocks',
+  		params: {size: this.state.pageSize, sort:'percentageChange,desc', page: this.state.page.number}
+  	}]).then(stockCollection => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/modernmaster/forecaster/pullRequest?prid=7238741)
```suggestion
  	}]).then((stockCollection) => {
```